samza-core/src/main/java/org/apache/samza/util/Util.java

```diff
@@ -45,7 +45,11 @@
    * Make an environment variable string safe to pass.
    */
   public static String envVarEscape(String str) {
-    return str.replace("\"", "\\\"").replace("'", "\\'");
+    return str
```

Comment:
Are these escape sequences comprehensive enough to cover all the special characters that are currently used in samza-sql statements? (Some other special characters include: `% `, `(`, `[`).

Reply:
%, ( and [ are not special characters in terms of env var.

Reply:
The point I am trying to make is passing samza-sql statements over as env_arguments is kind of fragile. To be comprehensive, you need to figure out all the special characters in bash that are used in sql and then escape them. Special characters that are used in [bash](https://docstore.mik.ua/orelly/unix/upt/ch08_19.htm)  .

Reply:
I understand your point. Special characters in bash does not mean we need to escape it because:
1. it must be valid SQL syntax
2. they are already encapsulated in bash string, only those that will impact the string encapsulation needs to be escaped. 

I can only find backslash and grave accent for now, do you know if there are other characters that we need to escape here?

Reply:
- From this bash wiki here: https://mywiki.wooledge.org/BashGuide/SpecialCharacters
`$`, `#`  are special charactors. They're used in samza-sql statements espciecially in udf's,   `( )` means sub-shell group in bash. It's also used in samza-sql.

- Moreover [join](https://shapeshed.com/unix-join/) is a linux command and samza-sql uses this keyword extensively for stream-table join and stream-to-stream join.   . Do you know if we had to escape that too.

Reply:
I may not be clear before, let me explain, all of these env var are already encapsulated in a string in bash, i.e. 
```
export SAMZA_SUBMISSION_CONFIG="the config"
```
what we need to escape are those characters that will change how bash interpret strings, single and double quotes are the most obvious ones, which are the characters we have been escaping. However, backslash and grave accent are also special characters that will change how string looks in bash, other special characters such as `$`, `#`, `(`, `)` are special in bash itself but no longer special once they are wrapper inside a string.